DoxygenRule.cmake

```diff
@@ -109,28 +109,26 @@ add_custom_target(doxygen_html_${PROJECT_NAME}
   COMMENT "Generating API documentation using doxygen" VERBATIM
   DEPENDS doxygen_install_${PROJECT_NAME} project_info_${PROJECT_NAME})
 set_target_properties(doxygen_html_${PROJECT_NAME} PROPERTIES
-  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
+  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "${PROJECT_NAME}/doxygen")
 
 if(NOT TARGET doxygen_html)
   add_custom_target(doxygen_html)
   set_target_properties(doxygen_html PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
+    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "${PROJECT_NAME}/doxygen")
 endif()
 add_dependencies(doxygen_html doxygen_html_${PROJECT_NAME})
 
 add_custom_target(${PROJECT_NAME}-doxygen DEPENDS doxygen_html_${PROJECT_NAME})
 set_target_properties(${PROJECT_NAME}-doxygen PROPERTIES
-  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
+  EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "${PROJECT_NAME}/doxygen")
 if(ENABLE_COVERAGE) # CoverageReport generated by "tests" in this case
   add_dependencies(${PROJECT_NAME}-doxygen coverage_${PROJECT_NAME})
 endif()
 
-if(NOT TARGET doxygen)
-  add_custom_target(doxygen)
-  set_target_properties(doxygen PROPERTIES
-    EXCLUDE_FROM_DEFAULT_BUILD ON FOLDER "doxygen")
+if(NOT TARGET doxygens)
+  add_custom_target(doxygens)
 endif()
-add_dependencies(doxygen ${PROJECT_NAME}-doxygen)
+add_dependencies(doxygens ${PROJECT_NAME}-doxygen)
```

Comment:
Oh, since you're changing this, what about naming the targets 'documentation' instead of 'doxygen'?